mono/mini/TestHelpers.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+
+namespace MonoTests.Helpers {
+
+	public static class FinalizerHelpers {
+		private static IntPtr aptr;
+
+		private static unsafe void NoPinActionHelper (int depth, Action act)
+		{
+			// Avoid tail calls
+			int* values = stackalloc int [20];
+			aptr = new IntPtr (values);
+
+			if (depth <= 0)
+				act ();
+			else
+				NoPinActionHelper (depth - 1, act);
+		}
+
+		/*
+		 * Returns an Action that calls the original action on the bottom of the stack
+		 * thus avoiding stack pinning of objects created by it.
+		 */
+		public static Action GetNoPinAction (Action act)
```

Comment:
Given we always what to start a new thread maybe we could go even further and do the whole thread start / join here as well?